rpi_backlight/cli.py

```diff
@@ -56,7 +67,21 @@ def main():
     parser = _create_argument_parser()
     args = parser.parse_args()
 
-    if args.sysfs_path is not None:
+    if args.board:
+        board = int(args.board)
+        if board != 1 and board != 2:
+            parser.error(
+                "Board parameter must be either 1 (Raspberry Pi) or 2 (TinkerBoard)"
+            )
+
+    if board == 2:
+        if args.sysfs_path is not None:
+            backlight = Backlight(
+                board_type=BoardType.TINKER_BOARD, backlight_sysfs_path=args.sysfs_path
+            )
+        elif args.sysfs_path is None:
+            backlight = Backlight(board_type=BoardType.TINKER_BOARD)
+    elif args.sysfs_path is not None:
         backlight = Backlight(backlight_sysfs_path=args.sysfs_path)
     else:
         backlight = Backlight()
```

Comment:
Now we can simplify this quite a bit:

```suggestion
    backlight = Backlight(
        board_type=board_types[args.board_type], backlight_sysfs_path=args.sysfs_path
    )
```

`board_types[args.board_type]` won't raise as the dict key will always be validated for use before this is reached; and `backlight_sysfs_path` now can be `None` and will be set automatically :+1:

Reply:
@linusg seems this is breaking the script
Traceback (most recent call last):
  File "/usr/local/bin/rpi-backlight", line 11, in <module>
    load_entry_point('rpi-backlight==2.0.1', 'console_scripts', 'rpi-backlight')()
  File "/usr/local/lib/python3.5/dist-packages/rpi_backlight-2.0.1-py3.5.egg/rpi_backlight/cli.py", line 71, in main
NameError: name 'board_types' is not defined

Reply:
Where are we defining board_types?